src/mito2/src/worker.rs

```diff
@@ -818,16 +820,24 @@ impl<S: LogStore> RegionWorkerLoop<S> {
                 WorkerRequest::EditRegion(request) => {
                     self.handle_region_edit(request).await;
                 }
-                // We receive a stop signal, but we still want to process remaining
-                // requests. The worker thread will then check the running flag and
-                // then exit.
                 WorkerRequest::Stop => {
                     debug_assert!(!self.running.load(Ordering::Relaxed));
                 }
-
                 WorkerRequest::SyncRegion(req) => {
                     self.handle_region_sync(req).await;
                 }
+                WorkerRequest::BulkInserts {
+                    metadata,
+                    request,
+                    sender,
+                } => {
+                    if let Some(region_metadata) = metadata {
+                        self.handle_bulk_inserts(request, region_metadata, write_requests, sender)
+                            .await;
+                    } else {
```

Comment:
How can a region metadata be none, while still handling its inserts? Seems to me an illegal state, better rule it out outside the region worker loop. Or maybe best pass the region id (and fetch the current region metadata inside)?